include/NetworkDevice.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <cstdint>
+#include <queue>
+
+enum class DeviceType: std::uint8_t {
+    Switch,
+    ServerSender,
+    ServerReceiver,
+};
+
+class NetworkDevice {
+public:
+    explicit NetworkDevice(DeviceType type);
+    virtual ~NetworkDevice() = default;
+
+    /**
+     * @brief Send events (i. e. packets) to the receiver node
+     *
+     * @details Receiver -- next device in the flow path
+     */
+    virtual void Send(std::uint32_t flow_id) = 0;
+
+    /**
+     * @brief Vector of pointers to adjacent network devices.
+     *
+     * @details Usage of std::weak_ptr is necessary to avoid cyclic links.
+     */
+    std::vector<std::weak_ptr<NetworkDevice>> adjacent_devices;
+
+    /**
+     * @brief Network device type.
+     */
+    DeviceType type;
+
+    /**
+     * @brief Device's current time in nanoseconds
+     */
+    std::uint64_t current_time_ns = 0;
```

Comment:
```
   /**
     * @brief Network device type.
     */
    DeviceType type;
```


```
   /**
     * @brief Device's current time in nanoseconds
     */
    std::uint64_t current_time_ns = 0;
```

This is self-explanatory. Comments to self-expanatory variables name are redundant, unless we are absolutely sure we want to go the Doxygen way (probably not, at this stage of the project).